src/main/kotlin/no/eirikb/avatest/actions/AvaJavaScriptTestRunnerRunConfigurationGenerator.kt

```diff
@@ -153,6 +154,14 @@ class AvaJavaScriptTestRunnerRunConfigurationGenerator : AnAction() {
             }
             node.workingDirectory = project.basePath
             node.inputPath = AppSettingsState.inputPath
+            if (node.inputPath == null) {
+                val projectDir = project.guessProjectDir()
+                node.inputPath = listOf(
+                    "node_modules/ava/cli.js",
+                    "node_modules/.bin/ava.cmd",
+                    "node_modules/.bin/ava"
+                ).find { projectDir?.findFileByRelativePath(it) != null }
```

Comment:
```suggestion
                if (projectDir == null) {
                    writeError("not found ava cli")
                    return null
                }

                var avaPath = File(projectDir.toString(), "node_modules")
                avaPath = File(avaPath, "ava")
                val avaPackageJson = File(avaPath, "package.json")
                
                // parse ava package.json
```

There are two options:

1. read `package.json` -> `bin` -> `ava key` -> `value`
2. if (`package.json` -> `version` > 3) { node.input = "node_modules/ava/entrypoints/cli.mjs" } else { node.input = "node_modules/ava/cli.js" }

**I don't know how to parse `package.json` in kt**. So I can only provide a small part of the code and ideas

Reply:
Additional information: 
we cannot use the scripts directly from .bin, this causes some problems

Reply:
That is an option, but would still need the check of `.cmd` to support Windows

Reply:
@eirikb at windows:

<img width="560" alt="图片" src="https://user-images.githubusercontent.com/8198408/146719701-151cdbb3-3e03-45a1-8e81-632dd488385b.png">

```shell
node node_modules/.bin/ava.cmd
// BOOM
```

<img width="1120" alt="图片" src="https://user-images.githubusercontent.com/8198408/146719905-55b23148-0daa-4089-8bf4-037e8989427b.png">

Reply:
> Additional information: we cannot use the scripts directly from .bin, this causes some problems

I did some tests and it worked on my machine. At least with version 4.
Debugging was a bit weird as I had to hit <kbd>F9</kbd> right after start, but then it worked.

The new file, ["lib/cli.js"](https://github.com/avajs/ava/blob/main/lib/cli.js) doesn't seem to support being started directly.
Could be I'm wrong, or there are other files we could use, but it didn't look like it.

Reply:
> @eirikb at windows:
> 
> <img alt="图片" width="560" src="https://user-images.githubusercontent.com/8198408/146719701-151cdbb3-3e03-45a1-8e81-632dd488385b.png">
> 
> ```shell
> node node_modules/.bin/ava.cmd
> // BOOM
> ```
> 
> <img alt="图片" width="1120" src="https://user-images.githubusercontent.com/8198408/146719905-55b23148-0daa-4089-8bf4-037e8989427b.png">

Is that really version 4?
Mine looks like this:
![image](https://user-images.githubusercontent.com/241706/146720162-979e60a0-3f45-46e8-9bd5-1707bfe0ac4c.png)

Reply:
Sorry. I have installed the wrong version, please wait. I will take a new screenshot

Reply:
I guess it depends on node version, I have **17.1.0** at the moment. Maybe we could generate a file like the one I have now and call that instead.

Reply:
ava version 4:

macOS `.bin/ava`
<img width="272" alt="图片" src="https://user-images.githubusercontent.com/8198408/146720441-70da2d4a-2039-4e77-a3eb-5d30a082249a.png">

Window `.bin/ava.cmd`
<img width="712" alt="图片" src="https://user-images.githubusercontent.com/8198408/146720526-bc82e7e1-c61e-4387-8d61-4ad513443ff2.png">

Linux:
<img width="763" alt="图片" src="https://user-images.githubusercontent.com/8198408/146720547-eb8ebf31-1fc3-45ca-bb95-5d2cedb293ed.png">

Reply:
With your existing code, I think you only need to change the listOf:

```
listOf(
    "node_modules/ava/lib/cli.js",
    "node_modules/ava/cli.js"
)
```

Reply:
Wait, we could use that file: https://github.com/avajs/ava/blob/main/entrypoints/cli.mjs
How about
```kotlin
listOf(
    "node_modules/ava/entrypoints/cli.mjs",
    "node_modules/ava/cli.js"
)
```

Reply:
Of course you can~

Reply:
That is what you said earlier, sorry, I didn't catch the file was different.
This is a new file for ava4, it doesn't exist for 3, and on my machine the .bin didn't use it so I never saw it.

Reply:
`findFileByRelativePath` should come with a function to detect if a file exists
so, there is nothing wrong with the way you code

Reply:
I'll just do that, with my check-thing. Using package.json might be better, and would work if they change it again, but my fix will probably be good enough